### ISSUE TITLE
How to get the audio stream during a real-time conversation?

### ISSUE THREAD

**Issue by 298029lkk** [siniypin/pjsip4net#109]
I want to get the voice of the other party in a stream.
Is there any good advice?

**Comment by siniypin** [comment]
Hello, 

you can do so by plugging your own sound port implementation to the conference bridge, more info here:
- https://trac.pjsip.org/repos/wiki/FAQ#audio-man (note sink only media ports);
- https://trac.pjsip.org/repos/wiki/media-flow

https://github.com/siniypin/pjsip4net/issues/77

P.S. If you end up developing one it would be really great if you could it share with everybody else.

**Comment by 298029lkk** [comment]
@siniypin Thank you for your reply. According to your suggestion, is it necessary to modify the source code of pjsip?

**Comment by siniypin** [comment]
No, you don't need to modify pjsip, but you certainly need to plug into pjsip audio pipeline using the above mentioned tricks.

**Comment by siniypin** [comment]
One hack I can think of that wouldn't require pjsip integration is to stream audio to a file while simultaneously reading from the very same file. Never tried it, not even sure it is going to work for you.

**Comment by 298029lkk** [comment]
Thank you for your support.
I want to get the other party's audio in real time after the phone is dialed in, then intelligently identify it, and then send it to the user. Instead of a person speaking at the end of the phone.
This is what I want to achieve now. The key point is that I can't handle the acquisition and sending of audio.

**Comment by siniypin** [comment]
Well, you've got these two options. You can route an audio stream to a file with pjsip4net using a [DSL builder method](https://github.com/siniypin/pjsip4net/blob/master/pjsip4net/Calls/Dsl/DefaultCallBuilder.cs#L55)

**Comment by 298029lkk** [comment]
Ok, thank you very much. But I think the second way can't achieve my purpose. Because I want a real-time conversation and a real-time answer. Instead of processing the recording file after the conversation is completed.
So should I use the first way? Where should the first way handle?
Also, if I have an audio stream answer now, how do I send it to the other party?

**Comment by siniypin** [comment]
I don't get what you aim for. It all sounds like you want to establish a common bi-directional call. pjsip4net does exactly that out of the box.
